code/modules/reagents/Chemistry-Goon-Medicine.dm

```diff
@@ -378,4 +378,218 @@ datum/reagent/morphine/on_mob_life(var/mob/living/M as mob)
 			M.drop_item()
 	cycle_count++
 	..()
-	return
\ No newline at end of file
+	return
+
+datum/reagent/oculine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	cycle_amount++
+	if(M.eye_blind > 0 && cycle_amount > 20)
+		if(prob(30))
+			M.eye_blind = 0
+		else if(prob(80))
+			M.eye_blind = 0
+			M.eye_blurry = 1
+		if(M.eye_blurry > 0)
+			if(prob(80))
+				M.eye_blurry = 0
+	..()
+	return
+
+/datum/chemical_reaction/oculine
+	name = "Oculine"
+	id = "oculine"
+	result = "oculine"
+	required_reagents = list("atropine" = 1, "spaceacillin" = 1, "salglu_solution" = 1)
+	result_amount = 3
+	mix_message = "The mixture sputters loudly and becomes a pale pink color."
+
+datum/reagent/oculine
+	name = "Oculine"
+	id = "oculine"
+	description = "30% chance to remove blindness, 80% chance to slightly reduce eye damage."
+	reagent_state = LIQUID
+	color = "#C8A5DC" // rgb: 200, 165, 220
+	metabolization_rate = 0.4
+	var/cycle_amount = 0
+
+datum/reagent/atropine
+	name = "Atropine"
+	id = "atropine"
+	description = "1 TOX damage if used over -60 health. Causes dizziness and confusion. If under -25 health, heals 3 BRUTE + 3 BURN. Attempts to cap OXY damage at 65 and LOSEBREATH at 5."
+	reagent_state = LIQUID
+	color = "#C8A5DC" // rgb: 200, 165, 220
+	metabolization_rate = 0.2
+
+datum/reagent/atropine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	if(M.health > -60)
+		M.adjustToxLoss(1*REM)
+	if(M.health < -25)
+		M.adjustBruteLoss(3*REM)
+		M.adjustFireLoss(3*REM)
+	if(M.oxyloss > 65)
+		M.setOxyLoss(65)
+	if(M.losebreath > 5)
+		M.losebreath = 5
+	if(prob(30))
+		M.Dizzy(5)
+		M.Jitter(5)
+	..()
+	return
+
+/datum/chemical_reaction/atropine
+	name = "Atropine"
+	id = "atropine"
+	result = "atropine"
+	required_reagents = list("ethanol" = 1, "acetone" = 1, "diethylamine" = 1, "phenol" = 1, "sacid" = 1)
+	result_amount = 5
+
+datum/reagent/epinephrine
+	name = "Epinephrine"
+	id = "epinephrine"
+	description = "Reduces most of the knockout/stun effects, minor stamina regeneration buff. Attempts to cap OXY damage at 35 and LOSEBREATH at 3. If health is between -10 to -65, heals 1 TOX, 1 BRUTE, 1 BURN."
+	reagent_state = LIQUID
+	color = "#C8A5DC" // rgb: 200, 165, 220
+	metabolization_rate = 0.2
+
+datum/reagent/epinephrine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	if(M.health > -10 && M.health < -65)
+		M.adjustToxLoss(-1*REM)
+		M.adjustBruteLoss(-1*REM)
+		M.adjustFireLoss(-1*REM)
+	if(M.oxyloss > 35)
+		M.setOxyLoss(35)
+	if(M.losebreath > 3)
+		M.losebreath = 3
+	M.adjustStaminaLoss(-1*REM)
+	if(prob(30))
+		M.AdjustParalysis(-1)
+		M.AdjustStunned(-1)
+		M.AdjustWeakened(-1)
```

Comment:
I would add some of these sound OP or will be a bit OP until their overdose and addiction side effects are completed.